### PR TITLE
eTransl text size limit to 5000, after that use a document-based

### PR DESCRIPTION
**Pull request by SrdjanStevanetic**
translation

### CODE REVIEW

File: translation-service-etranslation/src/main/java/eu/europeana/api/translation/service/etranslation/ETranslationTranslationService.java [hunk 1]
```diff
@@ -133,12 +139,23 @@ public void translate(List<TranslationObj> translationObjs) throws TranslationEx
...
       
     //create a redis message listener obj, and wait on that obj until it get notified from the redis publisher
-    createRedisMessageListenerAndWaitForResults(translationObjs, eTranslExtRef);
+    createRedisMessageListenerAndWaitForResults(translationObjs, eTranslExtRef, eTranslJointStr.length());
       
   }
   
-  private void createRedisMessageListenerAndWaitForResults(List<TranslationObj> translationObjs, String eTranslExtRef) throws TranslationException {
-    RedisMessageListener redisMessageListener = new RedisMessageListener();
+  private boolean sendRequestAsDocument(int textSize) {
```
Comment (gsergiu) [inline translation-service-etranslation/src/main/java/eu/europeana/api/translation/service/etranslation/ETranslationTranslationService.java]: please rename to isSnipetLimitExceeded()

File: translation-service-etranslation/src/main/java/eu/europeana/api/translation/service/etranslation/RedisMessageListener.java [hunk 2]
```diff
@@ -26,15 +33,20 @@ public void onMessage(Message message, byte[] pattern) {
...
+            this.message = messageBody;
+          }
+          else {
+            /* 
+             * the received message is treated as a json object and we need some adjustments for the escaped characters
+             * (this only applies if we get the translated text from the translated-text field in the eTransl callback,
+             * which happens if we send the text to be translated in the textToTranslate request param)
+             */
+            //remove double quotes at the beginning and at the end of the response, from some reason they are duplicated
+            String messageRemDuplQuotes = messageBody.replaceAll("^\"|\"$", "");
```
Comment (gsergiu) [inline translation-service-etranslation/src/main/java/eu/europeana/api/translation/service/etranslation/RedisMessageListener.java:46]: this doesn't look right, we need to investigate the issue and eventually report a bug to eTRansaltion

File: translation-web/src/main/java/eu/europeana/api/translation/web/service/TranslationWebService.java [hunk 3]
```diff
@@ -130,17 +135,98 @@ CachedTranslationService instantiateCachedTranslationService(boolean useCaching,
     }
   }
 
+  /*
+   * This method is used only for the purpose of eTranslation stress test and can be excluded afterwards
+   */
+  private void limitTextSizeForETranslationStressTest(TranslationRequest translationRequest, List<TranslationObj> translObjs) {
```
Comment (gsergiu) [inline translation-web/src/main/java/eu/europeana/api/translation/web/service/TranslationWebService.java:141]: this implementation looks overcomplicated, and it doesn't seem to take into account that we also append the delimiter

Comment (SrdjanStevanetic) [reply translation-web/src/main/java/eu/europeana/api/translation/web/service/TranslationWebService.java:141]: It takes into account the delimiter while calculating the length of the joint texts (so the length is with the delimiter), to be below the limit of 5000.

**Pull request merged**